### ISSUE TITLE
Python version support

### ISSUE THREAD

**Issue by rbeucher** [ACCESS-NRI/dev-docs#15]
I think we need to ensure that our Python packages support current versions of Python.

For example, Python is currently maintained for version 3.8 and above.
Versions 3.8, 3.9 and 3.10 only accept security fixes and are considered stable.
Version 3.11 only accepts bug fixes.

Version 3.12 is pre-release...
Version 3.13 is in dev.

Let us support code that runs on stable versions, meaning 3.8 to 3.10 at the moment.
People should avoid any new feature introduced in bugfix, prerelease and feature versions of Python.

I have added some comments to the WIKI.

@rhaegar325 @max-anu @SeanBryan51 @ccarouge

**Comment by rbeucher** [comment]
But as a rule. Our code should currently be Python 3.8 and above...

**Comment by rbeucher** [comment]
@dougiesquire any issue with the catalog regarding this?

**Comment by SeanBryan51** [comment]
I think supporting versions 3.8 and above sounds reasonable if we want our packages to be available via the `hh5` conda environment. I saw CLEX recently upgraded the `analysis3-unstable` environment to python 3.10.

**Comment by dougiesquire** [comment]
> @dougiesquire any issue with the catalog regarding this?

Don't think so. The catalog is currently [tested with 3.9, 3.10 and 3.11](https://github.com/ACCESS-NRI/access-nri-intake-catalog/blob/main/.github/workflows/ci.yml#L18).

**Comment by truth-quark** [comment]
Another consideration is that supported Python versions update on a near yearly basis, indicating ACCESS NRI will have to periodically cycle supported version numbers too.

**Comment by rbeucher** [comment]
Yes. 3.12 is now supported for example...

**Comment by aidanheerdegen** [comment]
I agree we should have an organisation policy on what our minimum supported python version is, and a process to keep it updated.
